### ISSUE TITLE
Cannot compile with ?Sized struct

### ISSUE THREAD

**Issue by taiki-e** [taiki-e/pin-project#112]
Code:

```rust
#[pin_project]
pub struct A<T: ?Sized> {
    x: T,
}
```

Error:

```text
the size for values of type `T` cannot be known at compilation time

doesn't have a size known at compile-time

help: within `dst::A<T>`, the trait `core::marker::Sized` is not implemented for `T`
note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
help: consider adding a `where T: core::marker::Sized` bound
note: required because it appears within the type `dst::A<T>`
note: all local variables must have a statically known size
help: unsized locals are gated as an unstable feature
```

**Comment by taiki-e** [comment]
This is because the `val` argument of `ensure_not_packed` function is passed by value.
https://github.com/taiki-e/pin-project/blob/6e3c1198e53c659ee795866b5017bd2d19676028/pin-project-internal/src/pin_project/attribute.rs#L384

**Comment by taiki-e** [comment]
Published 0.4.2 which fixes this issue.
